extension/lib/logger.js

```diff
@@ -108,24 +103,28 @@ Logger.prototype._writeAsync = function Logger_writeAsync(aMessage, aCallback) {
   // For testing purposes send the message to stdout
   //dump(aMessage + '\n');
 
-  // Create an output stream to write to file
-  var foStream = Cc["@mozilla.org/network/file-output-stream;1"]
-                 .createInstance(Ci.nsIFileOutputStream);
-  foStream.init(this._file, 0x02 | 0x08 | 0x10, PERMS_FILE, foStream.DEFER_OPEN);
-
-  // Write asynchronously to buffer;
-  // Input and output streams are closed after write
-  var iStream = this._converter.convertToInputStream(aMessage);
-  NetUtil.asyncCopy(iStream, foStream, function (status) {
-    if (!components.isSuccessCode(status)) {
-      var errorMessage = new Error("Error while writing to file: " + status);
-      console.error(errorMessage);
-    }
-
+  function callback(aResult) {
     if (typeof(aCallback) === "function") {
-      aCallback(status);
+      try {
+        aCallback(aResult);
+      } catch (e) {
+        Cu.reportError(e);
+      }
     }
-  });
+  }
+
+  let array = this._encoder.encode(aMessage);
+  let promise = OS.File.open(this.file.path, {write: true, append: true}).then(file => {
+    file.write(array).then(bytes => {
+      file.close();
+      callback(bytes);
+    },
+    error => {
+      file.close();
+      callback(error);
+    })
+  })
+  .catch(callback);
```

Comment:
Do we need the call to ´.catch()´ here? I actually have never seen that in other promise code yet. We have the ´error()` as last one. So does it not catch some of the exceptions?